Import mdadm-4.2-5.xs8.src.rpm

glehmann opened the pull request:
https://koji.xcp-ng.org/taskinfo?taskID=89509

rzr commented on SPECS/mdadm.spec:
```diff
@@ -0,0 +1,220 @@
+
+%changelog
+* Thu Aug 7 2025 Gaëtan Lehmann <gaetan.lehmann@vates.tech> - 4.2-5
+- Import XenServer's mdadm-4.2-5
```
Has an "openbase" considered before importing it  ?

glehmann replied:
openbase?

rzr replied:
i mean upstream package but looking at history, there are so many changes, maybe it is safer to do follow your strategy

rzr reviewed (approved):
Any opinion on the timer raid check service ?

ydirson reviewed (changes requested):
I'm very confused, the branch topology seems very wrong:
- there are 3 EL packages in various branches 
- the `upstream-8.3` and `XS-8.3` branches are not reachable from this PR branch
- the branch looking like an `XS-8.3` does not descend from an upstream import
- the merge commit brings a number of patchfiles not used in the specfile: this merge seems a mistake too, if we don't bring any change we should just use a pristine XS source package with no change of our own

rzr commented:
@ydirson good catch it looks like work was done twice, I just reset  branches, PR is matching the original XS branch which looked good:

- [x] Koji build Scratch (v8.3-incoming): https://koji.xcp-ng.org/taskinfo?taskID=98214

rzr commented on SOURCES/raid-check.timer:
```diff
@@ -0,0 +1,10 @@
+Description=Weekly RAID setup health check
+
+[Timer]
+OnCalendar=Sun *-*-* 01:00:00
```
Is it risky ?

rzr committed:
Import mdadm-4.2-8.fc40.src.rpm

Signed-off-by: Philippe Coval <philippe.coval@vates.tech>
Origin: https://archives.fedoraproject.org/pub/archive/fedora/linux/releases/40/Everything/source/tree/Packages/m/mdadm-4.2-8.fc40.src.rpm
Relate-to: XCPNG-1271
Origin: https://archives.fedoraproject.org/pub/archive/fedora/linux/releases/40/Everything/source/tree/Packages/m/mdadm-4.2-8.fc40.src.rpm
Relate-to: XCPNG-1271
Origin: https://archives.fedoraproject.org/pub/archive/fedora/linux/releases/40/Everything/source/tree/Packages/m/mdadm-4.2-8.fc40.src.rpm
Relate-to: XCPNG-1271

rzr committed:
Import mdadm-4.2-5.xs8.src.rpm

Signed-off-by: Philippe Coval <philippe.coval@vates.tech>
Origin: https://www.xenserver.com/#mdadm-4.2-5.xs8.src.rpm
Relate-to: XCPNG-1271

rzr commented on SPECS/mdadm.spec:
```diff
@@ -0,0 +1,222 @@
+Patch122: disable-Werror.patch
+
+BuildRequires: make
+BuildRequires: systemd-units binutils-devel gcc systemd-devel
```
Comparing to upstream:

```
-BuildRequires: systemd-rpm-macros binutils-devel gcc systemd-devel
-Requires: libreport-filesystem
-Requires(post): systemd coreutils
-Requires(preun): systemd
-Requires(postun): systemd coreutils
+BuildRequires: systemd-units binutils-devel gcc systemd-devel
+Requires(post): systemd-units coreutils
+Requires(preun): systemd-units
+Requires(postun): systemd-units coreutils

```
Ok for 1st change

```
No package systemd-rpm-macros available.
```


But is systemd-units  valuable ? 
```
rpm -q --whatprovides  systemd-units
systemd-219-57.5.xcpng8.3.x86_64
```

ydirson replied:
It is sure a strange idea to use `systemd-units` and not `systemd`, for getting those macros, though it does pull the same package.  Should we ever need to diverge from XS, we can think of putting this straight then.

rzr replied:
yes I believe there is no risk to align 

About the marcros it looks it was a subpackage introduced with systemd-217-4.fc22-337-gc9030f0

https://src.fedoraproject.org/rpms/systemd/c/c9030f04


It looks our centos base did not (despite later major version)

Import systemd-219-57.el7.centos.1.xs3.src.rpm

https://github.com/xcp-ng-rpms/systemd/blob/403febcd27fc503becb9b38e778cf5be96cf6e9c/SPECS/systemd.spec#L1660

It looks like a detached parallel developement

https://gitlab.com/CentOS/archives/git.centos.org/rpms/systemd/-/commit/d0811f4fabfb80b80df1c41aa43760bd9300a6be

rzr commented:
- [x] Koji build Scratch (v8.3-incoming): https://koji.xcp-ng.org/taskinfo?taskID=98233

ydirson commented on SPECS/mdadm.spec:
```diff
@@ -182,10 +181,6 @@ install -m 0644 %{SOURCE4} %{buildroot}%{_tmpfilesdir}/%{name}.conf
 mkdir -p %{buildroot}%{_localstatedir}/run/
 install -d -m 0710 %{buildroot}/run/%{name}/
 
-# abrt
-mkdir -p %{buildroot}/etc/libreport/events.d
-install -m644 %{SOURCE5} %{buildroot}/etc/libreport/events.d
-
```
```
Source5: mdadm_event.conf
```

This file's advertised role is "Save /proc/mdstat in case of crash in mdadm/mdmon".  Not having it may not be a big problem, but I kinda wonder why they decided to remove it.
This file comes from https://src.fedoraproject.org/rpms/mdadm/c/eb7fecee1121a2490109e6612804fe2f15a641cf
Digging the bz entry, this removal is surely linked to the "CP-46115: Remove libreport-filesystem dependency" change in XS' 4.2-3.  That sounds a bit unexpected, as the C7 RPM we're currently using actually ships this file, and is the sole user of that dependency we currently have.

I have no hard objection on taking this XS SRPM as is, but I kind of find this decision of theirs to disable a functionality present since 2012 a bit annoying.  Likely just because `mdadm` is a new package for them and they wanted to minimize the amount package they import.

rzr replied:
Sounds fair enough, too bad there is no feedback channel to ask confirmation, is there ?

ydirson reviewed (approved):
"yes but"

rzr commented:
- [x] Koji build (v8.3-incoming): https://koji.xcp-ng.org/taskinfo?taskID=98261#mdadm#da9e97e

rzr commented on SPECS/mdadm.spec:
```diff
@@ -0,0 +1,222 @@
+- CA-412437: mdadm-4.2-4 for XS8 and mdp fixes
+
+* Thu Aug 15 2024 Stephen Cheng <stephen.cheng@cloud.com> - 4.2-3
+- CP-46115: Remove libreport-filesystem dependency
```
Any idea on the motivation ?

Maybe this relates to:

https://github.com/xcp-ng-rpms/xenserver-status-report/blob/0b97f94b373a2568d2e3cc7744b7c4a431918b83/SPECS/xenserver-status-report.spec#L75

- CP-309293: Remove most dependencies on external programs

Update from @stormi:  we manage this differently using:

https://github.com/xcp-ng/xcp-ng-xapi-plugins/blob/master/SOURCES/etc/xapi.d/plugins/raid.py

So removal is not impacting for XCP-ng

cc: @ydirson @psafont

stormi replied:
> So removal is not impacting for XCP-ng

I don't guarantee anything about that. All I can say is that we have a plugin that XS doesn't have so it's good to check that it's not impacted. Whether anything else is impacted or not, I don't know.
